plugins/nanospell/plugin.js

```diff
@@ -819,12 +818,13 @@
 				if (blockList.length > 0) {
 					startCheckOrMarkWords(getUnknownWords(combinedText), blockList);
 				}
-				editor.getSelection().selectBookmarks(bookmarks); // we may not need to select, just blow it up
+
 			}
 
 			function getWords(block) {
 				var range = editor.createRange(),
 					currentWordObj,
+					bookmarks = editor.getSelection().createBookmarks2(false),
```

Comment:
bookmarks2 is a mutation insensitive bookmark (no span is created) therefore we can create it when scanning for words in order for the "don't check this current word" logic to work, without  restoring the selection to the bookmark afterwards.